src/modules/flight_mode_manager/tasks/Transition/FlightTaskTransition.cpp

```diff
@@ -88,122 +81,59 @@ bool FlightTaskTransition::update()
+	const Vector2f horizontal_acceleration_direction = Dcm2f(_yaw) * Vector2f(-1.0f, 0.0f);
+	_acceleration_setpoint.xy() = tanf(pitch_setpoint) * CONSTANTS_ONE_G * horizontal_acceleration_direction;
 
 	_yaw_setpoint = NAN;
 	return ret;
 }
 
-void FlightTaskTransition::updateParameters()
-{
-	// check for parameter updates
-	if (_parameter_update_sub.updated()) {
-		// clear update
-		parameter_update_s pupdate;
-		_parameter_update_sub.copy(&pupdate);
-
-		updateParametersFromStorage();
-	}
-}
-
-void FlightTaskTransition::updateParametersFromStorage()
-{
-	if (_param_handle_pitch_cruise_degrees != PARAM_INVALID) {
-		param_get(_param_handle_pitch_cruise_degrees, &_param_pitch_cruise_degrees);
-	}
-
-	if (_param_handle_vt_b_dec_i != PARAM_INVALID) {
-		param_get(_param_handle_vt_b_dec_i, &_param_vt_b_dec_i);
-	}
-
-	if (_param_handle_vt_b_dec_mss != PARAM_INVALID) {
-		param_get(_param_handle_vt_b_dec_mss, &_param_vt_b_dec_mss);
-	}
-}
-
-void FlightTaskTransition::updateSubscribers()
-{
-	_sub_vehicle_status.update();
-	_sub_position_sp_triplet.update();
-	_sub_vehicle_local_position.update();
-}
-
-bool FlightTaskTransition::isVtolFrontTransition() const
-{
-	return _sub_vehicle_status.get().in_transition_mode
-	       && _sub_vehicle_status.get().in_transition_to_fw;
-
-}
-
-bool FlightTaskTransition::isVtolBackTransition() const
-{
-	return _sub_vehicle_status.get().in_transition_mode
-	       && !_sub_vehicle_status.get().in_transition_to_fw;
-}
-
 float FlightTaskTransition::computeBackTranstionPitchSetpoint()
 {
-	const vehicle_local_position_s &local_pos = _sub_vehicle_local_position.get();
-	const position_setpoint_s &current_pos_sp = _sub_position_sp_triplet.get().current;
+	const Vector2f position_xy{_position};
+	const Vector2f velocity_xy{_velocity};
+	const Vector2f velocity_xy_direction = velocity_xy.unit_or_zero();
 
-	// Retrieve default decelaration setpoint from param
-	const float default_deceleration_sp = _param_vt_b_dec_mss;
+	float deceleration_setpoint = _param_vt_b_dec_mss;
 
-	float deceleration_sp = default_deceleration_sp;
+	if (_sub_position_sp_triplet.get().current.valid && position_xy.isAllFinite() && velocity_xy.isAllFinite()) {
+		Vector2f position_setpoint_local;
+		_geo_projection.project(_sub_position_sp_triplet.get().current.lat, _sub_position_sp_triplet.get().current.lon,
+					position_setpoint_local(0), position_setpoint_local(1));
 
-	const Vector2f position_local{local_pos.x, local_pos.y};
-	const Vector2f velocity_local{local_pos.vx, local_pos.vy};
-	const Vector2f acceleration_local{local_pos.ax, local_pos.ay};
+		const Vector2f pos_to_target = position_setpoint_local - position_xy; // backtransition end-point w.r.t. vehicle
+		const float dist_to_target_in_moving_direction = pos_to_target.dot(velocity_xy_direction);
 
-	const float accel_in_flight_direction = acceleration_local.dot(velocity_local.unit_or_zero());
```

Comment:
Looking back at this after the review comment the horizontal velocity estimate gets used here without checking validity. This probably didn't lead to big problems because `Vector2f(NAN, NAN).unit_or_zero()` == `Vector2f(0.f, 0.f)` but that's pretty implicit and not very clear 🤔

I didn't fix that fact but preserved as is in my cleanup.

Reply:
I added comments in bb3176706d6d427795bb3a6c82096eb057699365 why that works as is but it's to be addressed in another iteration with more behavioral changes.